cmd/immu/immu.go

```diff
@@ -750,33 +754,52 @@ secondRoot: %x at index: %d
 	cmd.AddCommand(man.Generate(cmd, "immu", "../docs/man/immu"))
 
 	if err := cmd.Execute(); err != nil {
-		_, _ = fmt.Fprintln(os.Stderr, err)
-		os.Exit(1)
+		c.QuitToStdErr(err)
 	}
 }
 
 func configureOptions(cmd *cobra.Command) error {
 	cmd.PersistentFlags().IntP("port", "p", gw.DefaultOptions().ImmudPort, "immudb port number")
 	cmd.PersistentFlags().StringP("address", "a", gw.DefaultOptions().ImmudAddress, "immudb host address")
 	cmd.PersistentFlags().StringVar(&o.CfgFn, "config", "", "config file (default path are config or $HOME. Default filename is immu.ini)")
+	cmd.PersistentFlags().BoolP("auth", "s", client.DefaultOptions().Auth, "use authentication")
 	if err := viper.BindPFlag("default.port", cmd.PersistentFlags().Lookup("port")); err != nil {
 		return err
 	}
 	if err := viper.BindPFlag("default.address", cmd.PersistentFlags().Lookup("address")); err != nil {
 		return err
 	}
+	if err := viper.BindPFlag("default.auth", cmd.PersistentFlags().Lookup("auth")); err != nil {
+		return err
+	}
 	viper.SetDefault("default.port", gw.DefaultOptions().ImmudPort)
 	viper.SetDefault("default.address", gw.DefaultOptions().ImmudAddress)
+	viper.SetDefault("default.auth", client.DefaultOptions().Auth)
 	return nil
 }
 
-func options(cmd *cobra.Command) (*client.Options, error) {
+func options(cmd *cobra.Command, withToken bool) (*client.Options, error) {
```

Comment:
Why here we need to add withToken? 
Simply if auth is enabled we could check for the token, without the boolean.

Reply:
@mmeloni The `withToken` value differs for each command as there are some commands which do not require authentication, like `login`, for example.

Reply:
But if login command send a token to the server token is simply ignored

Reply:
@mmeloni Ok, i removed it.